tests/Resources/ti.geolocation.test.js

```diff
@@ -303,16 +307,265 @@ describe.windowsBroken('Titanium.Geolocation', () => {
+						}
+						testCurrentHeading();
+					});
+				} else {
+					testCurrentHeading();
+				}
+			});
 		});
 
 		describe('#getCurrentPosition()', () => {
 			it('is a Function', () => {
 				should(Ti.Geolocation).have.a.property('getCurrentPosition').which.is.a.Function();
 			});
+
+			it('works via callback argument', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				// can't get permissions on macOS or actual iOS devices, since it prompts
+				if ((isMacOS || isIOSDevice) && !Ti.Geolocation.hasLocationPermissions(permission)) {
+					return finish(); // FIXME: How can we limit to ios only, and skip on macos?
+				}
+
+				Ti.Geolocation.accuracy = Ti.Geolocation.ACCURACY_HIGH;
+
+				function testCurrentPosition() {
+					Ti.Geolocation.getCurrentPosition(function (data) {
+						try {
+							should(data).have.property('success').which.is.a.Boolean();
+							if (OS_ANDROID && !data.success) {
+								// Sometimes fails on Android device/emulator with network/passive/gps is unavailable
+								should(data).have.property('code').which.is.not.eql(0);
+								should(data).have.property('error').which.match(/^\w+ is unavailable$/);
+							} else {
+								should(data).have.property('code').which.eql(0);
+								should(data.coords).be.an.Object();
+								should(data.coords).have.a.property('latitude').which.is.a.Number();
+								should(data.coords).have.a.property('longitude').which.is.a.Number();
+								should(data.coords).have.a.property('altitude').which.is.a.Number();
+								should(data.coords).have.a.property('accuracy').which.is.a.Number();
+								should(data.coords).have.a.property('heading').which.is.a.Number();
+								should(data.coords).have.a.property('speed').which.is.a.Number();
+								should(data.coords).have.a.property('timestamp').which.is.a.Number();
+							}
+						} catch (err) {
+							return finish(err);
+						}
+						finish();
+					});
+				}
+
+				if (!Ti.Geolocation.hasLocationPermissions(permission)) {
+					Ti.Geolocation.requestLocationPermissions(permission, function (e) {
+						if (!e.success) {
+							return finish(new Error('Failed to get Geolocation ALWAYS permission: ' + e.error));
+						}
+						testCurrentPosition();
+					});
+				} else {
+					testCurrentPosition();
+				}
+			});
+
+			it('works via Promise return value', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				// can't get permissions on macOS or actual iOS devices, since it prompts
+				if ((isMacOS || isIOSDevice) && !Ti.Geolocation.hasLocationPermissions(permission)) {
+					return finish(); // FIXME: How can we limit to ios only, and skip on macos?
+				}
+
+				Ti.Geolocation.accuracy = Ti.Geolocation.ACCURACY_HIGH;
+
+				function testCurrentPosition() {
+					const result = Ti.Geolocation.getCurrentPosition();
+					try {
+						result.should.be.a.Promise();
+					} catch (err) {
+						return finish(err);
+					}
+					result.then(data => {
+						try {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.geolocation.test.js;
  Line: 527;
-->

- :warning: tests/Resources/ti.geolocation.test.js line 527 – Each then() should return a value or throw (promise/always-return)